x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/stats/ForecastStats.java

```diff
@@ -0,0 +1,153 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.ml.stats;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A class to hold statistics about forecasts.
+ */
+public class ForecastStats implements ToXContentObject, Writeable {
+
+    public static class Fields {
+        public static final String TOTAL = "total";
+        public static final String JOBS = "jobs";
+        public static final String MEMORY = "memory_bytes";
+        public static final String RUNTIME = "processing_time_ms";
+        public static final String RECORDS = "records";
+        public static final String STATUSES = "status";
+    }
+
+    private long total;
+    private long jobsWithAtleastOneForecast;
+    private StatsAccumulator memoryStats;
+    private StatsAccumulator recordStats;
+    private StatsAccumulator runtimeStats;
+    private CountAccumulator statusCounts;
+
+    public ForecastStats() {
+        this.total = 0;
+        this.jobsWithAtleastOneForecast = 0;
+        this.memoryStats = new StatsAccumulator();
+        this.recordStats = new StatsAccumulator();
+        this.runtimeStats = new StatsAccumulator();
+        this.statusCounts = new CountAccumulator();
+    }
+
+    /*
+     * Construct ForecastStats for 1 job. Additional statistics can be added by merging other ForecastStats into it.
+     */
+    public ForecastStats(long total, StatsAccumulator memoryStats, StatsAccumulator recordStats, StatsAccumulator runtimeStats,
+            CountAccumulator statusCounts) {
+        this.total = total;
+        this.jobsWithAtleastOneForecast = total > 0 ? 1 : 0;
+        this.memoryStats = Objects.requireNonNull(memoryStats);
+        this.recordStats = Objects.requireNonNull(recordStats);
+        this.runtimeStats = Objects.requireNonNull(runtimeStats);
+        this.statusCounts = Objects.requireNonNull(statusCounts);
+    }
+
+    public ForecastStats(StreamInput in) throws IOException {
+        this.total = in.readLong();
+        this.jobsWithAtleastOneForecast = in.readLong();
+        this.memoryStats = new StatsAccumulator(in);
+        this.recordStats = new StatsAccumulator(in);
+        this.runtimeStats = new StatsAccumulator(in);
+        this.statusCounts = new CountAccumulator(in);
+    }
+
+    public void merge(ForecastStats other) {
+        if (other == null) {
+            return;
+        }
+        total += other.total;
+        if (other.total > 0) {
+            ++jobsWithAtleastOneForecast;
```

Comment:
`fs = new ForecastStats().merge(a.merge(b.merge(c.merge(someStats))))`
After this `fs. jobsWithAtleastOneForecast` would have a value of 1 regardless of what has been merged into should the value of `other. jobsWithAtleastOneForecast` be used or have I misunderstood this completely.

Reply:
I think you are right, it's not used like this right now, but that's a bad excuse for not fixing it. it should simply be `jobsWithAtleastOneForecast +=other.jobsWithAtleastOneForecast`, the if is superflous.